packages/block-library/src/gallery/deprecated.js

```diff
@@ -7,426 +7,581 @@ import { map, some } from 'lodash';
- * @return {number}           Default number of columns for the gallery.
+ * @return {number}            Default number of columns for the gallery.
  */
 export function defaultColumnsNumberV1( attributes ) {
 	return Math.min( 3, attributes.images.length );
 }
 
-const deprecated = [
-	{
-		attributes: {
-			images: {
-				type: 'array',
-				default: [],
-				source: 'query',
-				selector: '.blocks-gallery-item',
-				query: {
-					url: {
-						type: 'string',
-						source: 'attribute',
-						selector: 'img',
-						attribute: 'src',
-					},
-					fullUrl: {
-						type: 'string',
-						source: 'attribute',
-						selector: 'img',
-						attribute: 'data-full-url',
-					},
-					link: {
-						type: 'string',
-						source: 'attribute',
-						selector: 'img',
-						attribute: 'data-link',
-					},
-					alt: {
-						type: 'string',
-						source: 'attribute',
-						selector: 'img',
-						attribute: 'alt',
-						default: '',
-					},
-					id: {
-						type: 'string',
-						source: 'attribute',
-						selector: 'img',
-						attribute: 'data-id',
-					},
-					caption: {
-						type: 'string',
-						source: 'html',
-						selector: '.blocks-gallery-item__caption',
-					},
+/**
+ * Original function to determines new href and linkDestination values for an image block from the
+ * supplied Gallery link destination.
+ *
+ * Used in deprecations: v1-6.
+ *
+ * @param {Object} image       Gallery image.
+ * @param {string} destination Gallery's selected link destination.
+ * @return {Object}            New attributes to assign to image block.
+ */
+export function getHrefAndDestination( image, destination ) {
+	// Need to determine the URL that the selected destination maps to.
+	// Gutenberg and WordPress use different constants so the new link
+	// destination also needs to be tweaked.
+	switch ( destination ) {
+		case DEPRECATED_LINK_DESTINATION_MEDIA:
+			return {
+				href: image?.source_url || image?.url, // eslint-disable-line camelcase
+				linkDestination: LINK_DESTINATION_MEDIA,
+			};
+		case DEPRECATED_LINK_DESTINATION_ATTACHMENT:
+			return {
+				href: image?.link,
+				linkDestination: LINK_DESTINATION_ATTACHMENT,
+			};
+		case LINK_DESTINATION_MEDIA:
+			return {
+				href: image?.source_url || image?.url, // eslint-disable-line camelcase
+				linkDestination: LINK_DESTINATION_MEDIA,
+			};
+		case LINK_DESTINATION_ATTACHMENT:
+			return {
+				href: image?.link,
+				linkDestination: LINK_DESTINATION_ATTACHMENT,
+			};
+		case LINK_DESTINATION_NONE:
+			return {
+				href: undefined,
+				linkDestination: LINK_DESTINATION_NONE,
+			};
+	}
+
+	return {};
+}
+
+/**
+ * Gets an Image block from gallery image data
+ *
+ * Used to migrate Galleries to nested Image InnerBlocks.
+ *
+ * @param {Object} image    Image properties.
+ * @param {string} sizeSlug Gallery sizeSlug attribute.
+ * @param {string} linkTo   Gallery linkTo attribute.
+ * @return {Object}          Image block.
```

Comment:
Tiny nit: there's an additional white space character on the return type label
```suggestion
 * @return {Object}         Image block.
```